server/src/main/java/io/crate/planner/optimizer/rule/SwapHashJoin.java

```diff
@@ -70,6 +70,11 @@ public LogicalPlan apply(HashJoin plan,
                 plan.outputs()
             );
         }
-        return null;
```

Comment:
Doesn't returning null break out of the loop as `done` remains true?

https://github.com/crate/crate/blob/5b3107e3117700600298532ba685f0fac4af189a/server/src/main/java/io/crate/planner/optimizer/iterative/IterativeOptimizer.java#L130-L139

If returning `null` for "no change" isn't safe, we probably have more rules that would need to change, no?

Reply:
I think you are right, this change is not needed.